### DISCUSSION TITLE
Contributing to V6

### DISCUSSION THREAD

**Discussion by petebalsavias** [fabricjs/fabric.js#8751] [Q&A]
I would love to help contribute to testing, documentation, features as the project I'm working on really needs to utilize the new changes on groups.   However,  I'm really struggling to get an environment up and running with the source code.  

Would be great if someone could clarify a couple of items. 

Happy to open bugs on these - but I'm not sure they are actually bugs, and more likely a product of not understanding the development setup.

1) Should the latest from the master branch actually compile and build?  There are numerous issues in the branch from what I cn see.    Is there documentation on how this team is using tags and branches?    

2) Assuming the master branch isn't in a stable (or even buildable) state,  I've tried to checkout the v6.0.0-beta1 tag to a local branch and work with that version.  however, similar issues -- but mostly about the jsdom references.  Can someone please provide some details into how that should be setup locally?  

/index.ts → dist/index.mjs, dist/index.cjs, dist/index.min.js...
(!) Plugin typescript: @rollup/plugin-typescript TS7016: Could not find a declaration file for module 'jsdom/lib/jsdom/living/generated/utils.js'. '/Users/petebalsavias/Developer/javascript/sprout-studio/fabric.js/node_modules/jsdom/lib/jsdom/living/generated/utils.js' implicitly has an 'any' type.
  If the 'jsdom' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/jsdom'
src/env/node.ts: (3:20)

3 import utils1 from 'jsdom/lib/jsdom/living/generated/utils.js';
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by ShaMan123** [comment]
Hi!
Happy to hear. I started the same way. Needed it for a project.
I will do my best to help you contribute.
We've just added a section on setting up the testing env. Make sure you read [CONTRIBUTING](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
As a rule of thumb PRs don't get merged until all critical checks pass, the first of them being build.
Typescript warns many things, many are not important and some we still need to address. Don't pay too much attention to that right now.
Open issues so we can discuss what you found, understand if it is a bug or not and help you triage, fix, PR.
Once you ship your first PR the rest will fall into place.
Documentation and website are outdated. That effort will begin once v6 is officially released.

- Make sure you fetch upstream and update your origin
- Just in case reinstall deps
